Improve error message when pushing squashed branch to show human-readable warning

When pushing a squashed branch fails (if the branch had been pushed before), show a clear human-readable error instead of the raw `git` output and panic message:

```
2025/05/08 17:33:55 Execute: "gh" with args: "[pr create --title test(#131): refactor editor tests to use pipes and verify CLI output --body…
? Where should we push the '131-editor-test' branch? volodya-lombrozo/aidy

Creating pull request for 131-editor-test into main in volodya-lombrozo/aidy

To https://github.com/volodya-lombrozo/aidy.git
 ! [rejected]        HEAD -> 131-editor-test (non-fast-forward)
error: failed to push some refs to 'https://github.com/volodya-lombrozo/aidy.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
failed to run git: exit status 1
panic: exit status 1

goroutine 1 [running]:
github.com/volodya-lombrozo/aidy/output.(*editor).run(0xc000104480, {0xc0002dc180, 0x106})
	/Users/lombrozo/Workspace/OpenSource/aidy/output/output.go:93 +0x1ee
github.com/volodya-lombrozo/aidy/output.(*editor).Print(0xc000104480, {0xc0002de6c0?, 0x243d501?})
	/Users/lombrozo/Workspace/OpenSource/aidy/output/output.go:65 +0x369
main.pull_request({0x256aac0, 0xc0000125c0}, {0x256a010, 0xc00020a240}, {0x2568828, 0xc000096000}, {0x2568b18, 0xc00019e020}, {0x2565e38, 0xc000104480})
	/Users/lombrozo/Workspace/OpenSource/aidy/main.go:307 +0x48a
main.main()
	/Users/lombrozo/Workspace/OpenSource/aidy/main.go:105 +0xe2a
```

--- a/main.go
+++ b/main.go
@@ -104,7 +104,7 @@ func main() {
 	case "pr", "pull-request":
 		pull_request(gitService, aiService, gh, ch, out)
 	case "h", "heal":
-		heal(gitService, shell)
+		heal(gitService)
 	case "ci", "commit":
 		noAI := len(os.Args) > 2 && os.Args[2] == "-n"
 		commit(gitService, shell, noAI, aiService)
@@ -228,7 +228,10 @@ func issue(userInput string, aiService ai.AI, gh github.Github, ch cache.AidyCac
 		cmd = fmt.Sprintf("\n%s", escapeBackticks(fmt.Sprintf("gh issue create --title \"%s\" --body \"%s\"", healQuotes(title), healQuotes(body))))
 	}
 	cmd = fmt.Sprintf("%s%s\n", cmd, repo)
-	out.Print(cmd)
+    err = out.Print(cmd)
+    if err != nil {
+        log.Fatalf("Error during making an issue: %v", err)
+    }
 }
 
 func squash(gitService git.Git, shell executor.Executor, aiService ai.AI) {
@@ -271,7 +274,7 @@ func commit(gitService git.Git, shell executor.Executor, noAI bool, aiService ai
 			log.Fatalf("Error committing changes: %v", err)
 		}
 	}
-	heal(gitService, shell)
+	heal(gitService)
 }
 
 func pull_request(gitService git.Git, aiService ai.AI, gh github.Github, ch cache.AidyCache, out output.Output) {
@@ -304,10 +307,13 @@ func pull_request(gitService git.Git, aiService ai.AI, gh github.Github, ch cach
 	prtitle := healPRTitle(healQuotes(title), nissue)
 	prbody := healPRBody(healQuotes(body), nissue)
 	cmd := escapeBackticks(fmt.Sprintf("gh pr create --title \"%s\" --body \"%s\"%s", prtitle, prbody, repo))
-	out.Print(cmd)
+	err = out.Print(cmd)
+	if err != nil {
+		log.Fatalf("Error during making a pull-request: %v", err)
+	}
 }
 
-func heal(gitService git.Git, shell executor.Executor) {
+func heal(gitService git.Git) {
 	branchName, err := gitService.GetBranchName()
 	if err != nil {
 		log.Fatalf("Error getting branch name: %v", err)

--- a/main_test.go
+++ b/main_test.go
@@ -26,7 +26,7 @@ func TestHeal(t *testing.T) {
 	}
 	mockGit := &git.MockGit{Shell: mockExecutor}
 
-	heal(mockGit, mockExecutor)
+	heal(mockGit)
 
 	expectedCommands := []string{
 		"git commit --amend -m feat(#41): current commit message",

--- a/output/editor_test.go
+++ b/output/editor_test.go
@@ -1,6 +1,7 @@
 package output
 
 import (
+	"fmt"
 	"io"
 	"os"
 	"testing"
@@ -21,10 +22,29 @@ func TestEditor_Print_RunOption(t *testing.T) {
 	require.NoError(t, err, "failed to close write pipe")
 	command := "echo 'Hello, World!'"
 
-	editor.Print(command)
-
+	err = editor.Print(command)
+    
+	require.NoError(t, err, "Print should not return an error")
 	assert.Len(t, shell.Commands, 1, "expected 1 command to be run")
 	assert.Equal(t, "echo 'Hello, World!'", shell.Commands[0], "expected command to match")
+}
+
+func TestEditor_Print_RunOption_Error(t *testing.T) {
+	r, w, _ := os.Pipe()
+	shell := executor.NewMock()
+	shell.Err = fmt.Errorf("simulated error")
+	editor := NewEditor(shell)
+	editor.in = r
+	_, err := io.WriteString(w, "r\n")
+	require.NoError(t, err, "failed to write to pipe")
+	err = w.Close()
+	require.NoError(t, err, "failed to close write pipe")
+	command := "echo 'Hello, World!'"
+
+	err = editor.Print(command)
+
+	assert.Error(t, err, "expected an error when running the command")
+	assert.Equal(t, "simulated error", err.Error(), "expected error message to match")
 }
 
 func TestEditor_Print_PrintOption(t *testing.T) {
@@ -40,8 +60,9 @@ func TestEditor_Print_PrintOption(t *testing.T) {
 	require.NoError(t, err, "failed to close write pipe")
 	command := "echo 'Hello, World!'"
 
-	editor.Print(command)
+	err = editor.Print(command)
 
+	assert.NoError(t, err, "Print should not return an error")
 	err = output_w.Close()
 	require.NoError(t, err, "failed to close output pipe")
 	output, err := io.ReadAll(output_r)
@@ -63,7 +84,7 @@ func TestEditor_Print_CancelOption(t *testing.T) {
 	require.NoError(t, err, "failed to close write pipe")
 	command := "echo 'Hello, World!'"
 
-	editor.Print(command)
+	_ = editor.Print(command)
 
 	err = output_w.Close()
 	require.NoError(t, err, "failed to close output pipe")
@@ -82,9 +103,9 @@ func TestEditor_Print_EditOption(t *testing.T) {
 	require.NoError(t, err, "failed to write to pipe")
 	err = w.Close()
 	require.NoError(t, err, "failed to close write pipe")
-
 	command := "echo 'Hello, World!'"
-	editor.Print(command)
+
+	_ = editor.Print(command)
 
 	assert.Len(t, shell.Commands, 2, "expected 2 commands to be run")
 	assert.Equal(t, command, shell.Commands[1], "expected edited command to match")

--- a/output/output.go
+++ b/output/output.go
@@ -12,7 +12,7 @@ import (
 )
 
 type Output interface {
-	Print(command string)
+	Print(command string) error
 }
 
 type editor struct {
@@ -48,7 +48,7 @@ func NewMock() *Mock {
 	return &Mock{}
 }
 
-func (e *editor) Print(command string) {
+func (e *editor) Print(command string) error {
 	cmd := prettyCommand(command)
 	fmt.Printf("\ngenerated command:\n%s\n", cmd)
 	for {
@@ -57,40 +57,40 @@ func (e *editor) Print(command string) {
 		line, err := reader.ReadString('\n')
 		if err != nil {
 			e.printfErr("%s: %v\n", "Error reading input", err)
-			return
+			return err
 		}
 		choice := strings.ToLower(strings.TrimSpace(line))
 		switch choice {
 		case "r":
-			e.run(cmd)
-			return
+			return e.run(cmd)
 		case "e":
 			updated := e.edit(cmd)
 			if updated == "" {
-				return
+				return nil
 			}
 			cmd = updated
-			e.run(cmd)
-			return
+			return e.run(cmd)
 		case "c":
 			e.printf("%s\n", "canceled.")
-			return
+			return nil
 		case "p":
 			e.printf("%s\n", cmd)
-			return
+			return nil
 		default:
 			e.printfErr("%s\n", "please type r, e, c, or p and press enter.")
 		}
 	}
 }
 
-func (e *editor) run(command string) {
+func (e *editor) run(command string) error {
 	e.printf("running '%s' command\n", command)
 	parts := cleanQoutes(splitCommand(command))
 	e.printf("command parts '%v'\n", parts)
 	_, err := e.shell.RunInteractively(parts[0], parts[1:]...)
 	if err != nil {
-		panic(err)
+		return err
+	} else {
+		return nil
 	}
 }
 
@@ -163,12 +163,14 @@ func prettyCommand(command string) string {
 	return res.String()
 }
 
-func (p *printer) Print(command string) {
+func (p *printer) Print(command string) error {
 	fmt.Println(command)
+	return nil
 }
 
-func (m *Mock) Print(command string) {
+func (m *Mock) Print(command string) error {
 	m.captured = append(m.captured, command)
+	return nil
 }
 
 func (m *Mock) Last() string {

--- a/output/output_test.go
+++ b/output/output_test.go
@@ -79,28 +79,26 @@ func TestFindEditor(t *testing.T) {
 
 func TestMockOutput(t *testing.T) {
 	mock := NewMock()
-
-	// Test capturing a single command
-	mock.Print("command1")
+	_ = mock.Print("command1")
 	require.Equal(t, "command1", mock.Last(), "Last command should be 'command1'")
 
-	// Test capturing multiple commands
-	mock.Print("command2")
+	_ = mock.Print("command2")
 	require.Equal(t, "command2", mock.Last(), "Last command should be 'command2'")
 
-	// Test capturing another command
-	mock.Print("command3")
+	_ = mock.Print("command3")
 	require.Equal(t, "command3", mock.Last(), "Last command should be 'command3'")
-
-	// Test panic when no commands are captured
-	defer func() {
-		if r := recover(); r == nil {
-			t.Errorf("Expected panic when no commands are captured")
-		}
-	}()
-	emptyMock := NewMock()
-	emptyMock.Last()
 }
+
+func TestMockOutput_Panics_WhenNoCommands(t *testing.T) {
+    mock := NewMock()
+    defer func() {
+        if r := recover(); r == nil {
+            t.Errorf("Expected panic when no commands are captured")
+        }
+    }()
+    mock.Last()
+}
+
 func TestCleanQoutes(t *testing.T) {
 	tests := []struct {
 		input    []string
@@ -127,15 +125,12 @@ func TestPrinter_Print(t *testing.T) {
 	os.Stdout = w
 	printer := NewPrinter()
 
-	printer.Print("hello")
-
-	if err := w.Close(); err != nil {
-		t.Errorf("failed to close write pipe: %v", err)
-	}
+	_ = printer.Print("hello")
+    err := w.Close();
+    require.NoError(t, err, "failed to close write pipe")
 	var buf bytes.Buffer
-	if _, err := io.Copy(&buf, r); err != nil {
-		t.Errorf("failed to copy data: %v", err)
-	}
+    _, err = io.Copy(&buf, r);
+    require.NoError(t, err, "failed to copy data")
 	os.Stdout = originalStdout
 	expected := "hello\n"
 	require.Equal(t, expected, buf.String(), "Output should match expected value")

